mindsdb/integrations/utilities/rag/splitters/file_splitter.py

```diff
@@ -2,80 +2,109 @@
 from typing import Callable, List, Union
 
 from langchain_core.documents import Document
-from langchain_text_splitters import MarkdownHeaderTextSplitter, HTMLHeaderTextSplitter, RecursiveCharacterTextSplitter
+from langchain_text_splitters import (
+    MarkdownHeaderTextSplitter,
+    HTMLHeaderTextSplitter,
+    RecursiveCharacterTextSplitter,
+)
+
+from mindsdb.interfaces.knowledge_base.preprocessing.models import TextChunkingConfig
 
 from mindsdb.utilities import log
 
 DEFAULT_CHUNK_SIZE = 1000
 DEFAULT_CHUNK_OVERLAP = 50
 DEFAULT_MARKDOWN_HEADERS_TO_SPLIT_ON = [
-    ('#', 'Header 1'),
-    ('##', 'Header 2'),
-    ('###', 'Header 3')
```

Comment:
nit: would rather put formatting changes in a separate PR to make it clearer what functionality is being change